src/protocols/ssh/ttymode.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "config.h"
+#include "ttymode.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+guac_ssh_ttymodes init_ttymodes() {
+    // Simple allocation for a placeholder
+    guac_ssh_ttymode* ttymode_array = malloc(1);
+    
+    // Set up the initial data structure.
+    guac_ssh_ttymodes empty_modes = {
+        ttymode_array,
+        0
+    };
+
+    return empty_modes;
```

Comment:
Assigning/returning a struct by value like this actually does an implicit copy in C.

Reply:
So, I take it this is bad?  Would it be preferable to pass by reference or something like that?